src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -381,7 +386,23 @@ public SCMRevision run(GitClient client, String remoteName) throws IOException,
                                           // tag does not exist
                                           return null;
                                       }
+                                  } else if (head instanceof GitBranchSCMHead) {
+                                      for (Branch b : client.getRemoteBranches()) {
+                                          String branchName = StringUtils.removeStart(b.getName(), remoteName + "/");
+                                          if (branchName.equals(head.getName())) {
+                                              return new GitBranchSCMRevision((GitBranchSCMHead)head, b.getSHA1String());
+                                          }
+                                      }
+                                  } else if (head instanceof GitRefSCMHead) {
+                                      try {
+                                          ObjectId objectId = client.revParse(((GitRefSCMHead) head).getRef());
```

Comment:
hmmm if this works, do we need the special case for `GitBranchSCMHead`? can we not just have this case for `head instanceof GitSCMHeadMixin`? (BTW fine if you don't want to do that... but would seem a more efficient method... in fact we probably could do the revParse for the default code path)

Reply:
That lead me to start searching for tests touching this code and sprinkle in some `GitBranchSCMHead`s in place of the original `SCMHead` and now I'm in a debugging rabbithole :)

Reply:
what was the outcome here?

Reply:
There was some back and forth discussions and some misunderstandings on my part.
I wrote some tests to target this, but I don't know how well it covers it, so some exploratory testing was also done.

IIRC In the end with regards to code review I though that what you were after was covered by the `else` below, but you were reviewing from your phone and said you would take a look at it later.

Reply:
yes covered by the else... i think I was just wondering if we could merge one of the paths... no worries. fine for now